Modules/_csv.c

```diff
@@ -377,7 +409,7 @@ dialect_new(PyTypeObject *type, PyObject *args, PyObject *kwargs)
 
     self = (DialectObj *)type->tp_alloc(type, 0);
     if (self == NULL) {
-        Py_XDECREF(dialect);
+        Py_CLEAR(dialect);
```

Comment:
Technically, Py_XDECREF is just fine, there is no need to set dialect to NULL, since the dialect local scope if this function, and we exit the scope (function) at the next line. Same remark for the "err" label. But it's up to you, it's just a remark ;-)

Reply:
I'll just leave that up to the compiler :)